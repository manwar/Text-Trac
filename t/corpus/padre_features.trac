[[PageOutline(1-2)]]

List of existing and planned features with links to description of each feature
with relevant tickets and test cases to make sure the features are implemented correctly.

= Features =

 * multi-platform: [wiki:DownloadWindows Microsoft Windows], [wiki:Download Linux], [wiki:DownloadFreeBSD FreeBSD], [wiki:DownloadMacOSX Mac OS X], [wiki:DownloadNetBSD NetBSD]
 * Free and Open Source Software under the Perl licensing terms: [http://svn.perlide.org/padre/trunk/Padre/COPYING GPL I] [http://svn.perlide.org/padre/trunk/Padre/Artistic Artistic I]
 * Written in Perl. There are parts that rely on libraries written in C or C++ but they are all wrapped in CPAN packages and have an API in Perl 5.
 * [wiki:Features/BasicEditorFeatures Basic Editor Features]
 * [wiki:Features/VisualEditorEffects Visual Editor Effects]
   * Line Numbers
   * Code Folding
   * Current Line
   * Right Margin
   * Newlines
   * Whitespaces
   * Indentation Guide 
 * [wiki:Features/SyntaxHighlighting Syntax Highlighting]
 * [wiki:Features/Run Run a Script]
 * [wiki:Features/TitleAndStatusBarPersonalization Title and Status Bar Personalization]
 * [wiki:Features/CommandLineWindow Command Line Window]
 * [wiki:Features/OpenResource Open Resource]
 * [wiki:Features/QuickMenuAccess Quick Menu Access]
 * [wiki:Features/SingleInstance Single Instance]
 * [wiki:Features/AutoSave Automatic Save and Backup]
 * [wiki:Features/RemoteEditing Remote Editing via FTP, HTTP, SSH]
 * [wiki:Features/CooperativeEditing Cooperative Editing] (Swarm)
 * [wiki:Features/SearchAndReplace Search and Replace]
 * [wiki:Features/IncrementalSearch Incremental Search]
 * [wiki:Features/FindInFiles Find In Files]
 * [wiki:Features/TODOList TODO List]
 * [wiki:Features/EditPatch Patch]
 * [wiki:Features/AutoDiff Auto-Diff]
 * [wiki:Features/CodeSnippets Code Snippets]
 * [wiki:Features/CPANModuleInstaller CPAN Module Installer]
 * [wiki:Features/SessionManagement Session Management]
 * [wiki:Features/SplitView Split View]
 * [wiki:Features/Bookmarks Bookmarks]
 * [wiki:Features/RightClick Context Menu]
 * [wiki:Features/ProjectManagement Project Management]
 * [wiki:Features/ProjectSkeletonGeneration Project Skeleton Generation]
 * [wiki:Features/Autocomplete Autocomplete]
 * [wiki:Features/HelpViewer Help Viewer]
 * [wiki:Features/ContextSensitiveHelp Context Sensitive Help]
 * [wiki:Features/RectangularEditing Rectangular Selection and Editing]
 * [wiki:Features/Autoindent Autoindent]
 * [wiki:Features/Brackets Bracket Completition and Matching]
 * [wiki:Features/Outline Code Browser and Var tree called Outline] 
 * [wiki:Features/Autocomplete Source code suggestions and Autocomplete]
 * [wiki:Features/PreferenceSystem Configuration and Preference System]
 * [wiki:Features/OpenSelection Open Selection]
 * Multi-document editing - as far as we understand this only means you can edit several files at the same time
 * Code-assist editor - What does this cover?
 * IME - [wiki:Features/MultilingualInputMethodEditor Multilingual Input Method Editor]
 * [wiki:Features/IntegratedWebServer Integrated Web Server] for web application development
 * Call tips
* [wiki:Features/ToolsPrefernces Tools Prefernces draft]

== Perl-specific features ==

 * [wiki:Features/SyntaxChecking Real Time Syntax Checking For Perl 5 and 6]
 * [wiki:Features/Perl5Debugger Perl 5 Debugger] - perl debugging in Padre
 * [wiki:Features/RefactoringPerl5 Refactoring Perl 5 code]
 * [wiki:Features/AutomatedBuildAndTest Automated Build and Test]
 * [wiki:Features/Perl6 Perl 6 features]
 * [wiki:Features/RegexEditor Regex Editor]
 * [wiki:PluginAPI Extendable in Perl 5] via a Powerful [wiki:PluginAPI Plugin API]
 * [wiki:Features/PerlTidy] - automatic reformatting (tidying) of code
 * [wiki:Features/REPL REPL]
 * [wiki:Features/ProjectBrowser Project Browser] (Tree view of source files and resources)
 * [wiki:Features/IdentifyCommonErrors Identify Common Errors]
 * special features for beginners (we can create a page listing them)
 * advanced features (we can create a page listing them)

 * [wiki:Features/Extendable Extendable in Perl 6 and other languages]
 * [wiki:Features/HandleMultipleLanguages]


== Plugins ==

 * [wiki:Plugins] generic plugins related tickets
 * [wiki:Features/IntegratedVersionControl Integrated Version Control]
 * [wiki:Features/SuperGrep Super Grep]

= Feature requests and ideas from other sources =

In this section we have lists of features taken from verious articles
and web sites comparing tools and even from other IDEs.

For each feature we first link it to a related ticket but later
on we will point to the description of our implementation of
that feature.

== From Perlmonks ==

Perlmonks has a table comparing features of IDEs and editors to be used for Perl
development.

The table can be found at [http://www.perlmonks.org/?node=Perl%20Development%20Tools Perl Development Tools]

TODO: add links to the relevant Feature Pages

|| '''Feature'''      || '''Status or link to ticket''' ||
|| Perl Exclusive     || Why is yes considered good here? ||
|| Free               || yes - Under the "perl" license. ||
|| Perl Engine        || Any - Padre itself runs on 5.8.7 or higher but your code can run on any version of perl ||
|| [wiki:Features/SyntaxChecking Syntax Checking]  || yes ||
|| [wiki:Features/SyntaxHighlighting Syntax colouring]  || yes ||
|| [wiki:Features/Autoindent Autoindent]    || yes ||
|| [wiki:Features/IncrementalSearch Incremental search] || yes ||
|| [wiki:Features/Brackets Bracket Completition and Matching] || yes ||
|| Folding            || see under [wiki:Features/VisualEditorEffects Visual Editor Effects] ||
|| Var tree           || see in [wiki:Features/Outline Outline] ||
|| Code browser       || see in [wiki:Features/Outline Outline] ||
|| Debugger           || [wiki:Features/Perl5Debugger Perl 5 Debugger] ||
|| Stepping           || See Debugger ||
|| Breakpoints        || See Debugger ||
|| Watches            || See Debugger ||
|| Var evaluation     || See Debugger ||
|| Stack module       || See Debugger ||
|| [wiki:Features/Bookmarks Bookmarks]    || yes ||
|| Suggest & complete || see #67 but add more ||
|| [wiki:Features/IdentifyCommonErrors Identify errors] || also [wiki:Features/SyntaxChecking Syntax Checking] ||
|| [wiki:Features/REPL Interactive shell]  || ||
|| [wiki:Features/CodeSnippets Code snippets]  || yes ||
|| [wiki:Features/ProjectManagement Project] || ||
|| Control version    ||  [wiki:Features/IntegratedVersionControl Integrated Version Control] ||
|| Multi document     || yes ||
|| [wiki:Features/SplitView Split View] || ||
|| IME                || Multilingual Input Method Editor ||
|| [wiki:Features/HelpViewer Docs & Help viewer] || yes ||
|| [wiki:Features/CPANModuleInstaller Module installer]   || ||
|| WEB                || [wiki:Features/IntegratedWebServer Integrated Web Server] ||
|| FTP                || #12 Read and write over ftp protocol (modify the program at remote server) ||
|| HTML export        || #112 To export the source of program with the syntax colouring. ||
|| Linux              || [wiki:Download] ||
|| BSD                || [wiki:DownloadFreeBSD] and [wiki:DownloadNetBSD] ||
|| Mac OS X           || [wiki:DownloadMacOSX] ||
|| Solaris            || ||
|| Windows            || [wiki:DownloadWindows] ||


== From Perl Needs Better Tools ==

[http://www.eigenstate.net/perl_tools/ Perl Needs Better Tools] by Matisse Enzer 

||'''Feature'''                                                    || '''Status or link to ticket''' ||
||[wiki:Features/SyntaxHighlighting Syntax coloring]               || ||
||[wiki:Features/SyntaxChecking Real Time Syntax Checking] || ||
||[wiki:Features/IntegratedVersionControl Version control integration] || ||
||Code-assist editor                                               ||  ||
||Excellent refactoring support                                    || [wiki:Features/RefactoringPerl5 Refactoring Perl 5 code] ||
||Tree view of source files and resources                          ||  ||
||Support for creating and running unit tests                      || [wiki:Features/AutomatedBuildAndTest Automated Build and Test] ||
||Language-specific help                                           || [wiki:Features/ContextSensitiveHelp Context Sensitive Help] ||
||Run your code under a debugger with real-time display of results ||  ||
||Automatic reformatting of code.                                  ||  ||
||handling of multiple languages                                   ||  ||
||Automated build and test                                         ||  ||


== From Enlightened Perl ==

Work together with the folks at Enlightened Perl http://www.enlightenedperl.org/
to see how Padre can provide a GUI for their efforts.
Look at Kensho, http://search.cpan.org/dist/Task-Kensho/ a list of recommended module
and see how Padre can provide a GUI (via a plugin or in core Padre) for them.

== From Competitors ==

See the [wiki:'Competitor Analysis']

== From Eclipse EPIC ==

[http://www.eclipse.org/cdt/ Eclipse CDT - C/C++ Development Tooling]
[http://www.eclipse.org/jdt/ Eclipse Java development tools (JDT)]

Some of these features are only relevant (or possible) for C/C++ or Java

||'''Feature'''                                            || '''Status or link to ticket''' ||
||Find all references                                      || ||
||Find definition/declaration/usage                        || ||
||Type Hierarchy                                           || ||
||Call Hierarchy (or call graph)                           || ||
||[wiki:Features/RefactoringPerl5 Refactoring]             || ||
||[wiki:Features/SourceNavigation Source navigation]       || ||
||[wiki:Features/SyntaxHighlighting Syntax highlighting]   || ||
||Code folding                                             || under [wiki:Features/VisualEditorEffects Visual Editor Effects] ||
||Hyperlink navigation                                     || ||
||Include browser                                          || ||
||Project creation                                         || [wiki:Features/ProjectSkeletonGeneration Project Skeleton Generation] ||
||Managed build for various toolchains                     || ||
||Standard make build                                      || Via Run menu ||
||Macro definition browser                                 || ||
||Code generation                                          || partly (File -> New...) ||
||Visual debugging tools                                   || ||
||Memory, registers, and disassembly viewers               || ||


