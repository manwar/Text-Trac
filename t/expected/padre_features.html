<p>
[[PageOutline(1-2)]]
</p>
<p>
List of existing and planned features with links to description of each feature
with relevant tickets and test cases to make sure the features are implemented correctly.
</p>
<h1 id="Features">Features</h1>
<ul><li>multi-platform: <a class="wiki" href="/wiki/DownloadWindows">Microsoft Windows</a>, <a class="wiki" href="/wiki/Download">Linux</a>, <a class="wiki" href="/wiki/DownloadFreeBSD">FreeBSD</a>, <a class="wiki" href="/wiki/DownloadMacOSX">Mac OS X</a>, <a class="wiki" href="/wiki/DownloadNetBSD">NetBSD</a>
</li><li>Free and Open Source Software under the Perl licensing terms: <a class="ext-link" href="http://svn.perlide.org/padre/trunk/Padre/COPYING"><span class="icon"></span>GPL I</a> <a class="ext-link" href="http://svn.perlide.org/padre/trunk/Padre/Artistic"><span class="icon"></span>Artistic I</a>
</li><li>Written in Perl. There are parts that rely on libraries written in C or C++ but they are all wrapped in CPAN packages and have an API in Perl 5.
</li><li><a class="wiki" href="/wiki/Features/BasicEditorFeatures">Basic Editor Features</a>
</li><li><a class="wiki" href="/wiki/Features/VisualEditorEffects">Visual Editor Effects</a>
<ul><li>Line Numbers
</li><li>Code Folding
</li><li>Current Line
</li><li>Right Margin
</li><li>Newlines
</li><li>Whitespaces
</li><li>Indentation Guide 
</li></ul></li><li><a class="wiki" href="/wiki/Features/SyntaxHighlighting">Syntax Highlighting</a>
</li><li><a class="wiki" href="/wiki/Features/Run">Run a Script</a>
</li><li><a class="wiki" href="/wiki/Features/TitleAndStatusBarPersonalization">Title and Status Bar Personalization</a>
</li><li><a class="wiki" href="/wiki/Features/CommandLineWindow">Command Line Window</a>
</li><li><a class="wiki" href="/wiki/Features/OpenResource">Open Resource</a>
</li><li><a class="wiki" href="/wiki/Features/QuickMenuAccess">Quick Menu Access</a>
</li><li><a class="wiki" href="/wiki/Features/SingleInstance">Single Instance</a>
</li><li><a class="wiki" href="/wiki/Features/AutoSave">Automatic Save and Backup</a>
</li><li><a class="wiki" href="/wiki/Features/RemoteEditing">Remote Editing via FTP, HTTP, SSH</a>
</li><li><a class="wiki" href="/wiki/Features/CooperativeEditing">Cooperative Editing</a> (Swarm)
</li><li><a class="wiki" href="/wiki/Features/SearchAndReplace">Search and Replace</a>
</li><li><a class="wiki" href="/wiki/Features/IncrementalSearch">Incremental Search</a>
</li><li><a class="wiki" href="/wiki/Features/FindInFiles">Find In Files</a>
</li><li><a class="wiki" href="/wiki/Features/TODOList">TODO List</a>
</li><li><a class="wiki" href="/wiki/Features/EditPatch">Patch</a>
</li><li><a class="wiki" href="/wiki/Features/AutoDiff">Auto-Diff</a>
</li><li><a class="wiki" href="/wiki/Features/CodeSnippets">Code Snippets</a>
</li><li><a class="wiki" href="/wiki/Features/CPANModuleInstaller">CPAN Module Installer</a>
</li><li><a class="wiki" href="/wiki/Features/SessionManagement">Session Management</a>
</li><li><a class="wiki" href="/wiki/Features/SplitView">Split View</a>
</li><li><a class="wiki" href="/wiki/Features/Bookmarks">Bookmarks</a>
</li><li><a class="wiki" href="/wiki/Features/RightClick">Context Menu</a>
</li><li><a class="wiki" href="/wiki/Features/ProjectManagement">Project Management</a>
</li><li><a class="wiki" href="/wiki/Features/ProjectSkeletonGeneration">Project Skeleton Generation</a>
</li><li><a class="wiki" href="/wiki/Features/Autocomplete">Autocomplete</a>
</li><li><a class="wiki" href="/wiki/Features/HelpViewer">Help Viewer</a>
</li><li><a class="wiki" href="/wiki/Features/ContextSensitiveHelp">Context Sensitive Help</a>
</li><li><a class="wiki" href="/wiki/Features/RectangularEditing">Rectangular Selection and Editing</a>
</li><li><a class="wiki" href="/wiki/Features/Autoindent">Autoindent</a>
</li><li><a class="wiki" href="/wiki/Features/Brackets">Bracket Completition and Matching</a>
</li><li><a class="wiki" href="/wiki/Features/Outline">Code Browser and Var tree called Outline</a> 
</li><li><a class="wiki" href="/wiki/Features/Autocomplete">Source code suggestions and Autocomplete</a>
</li><li><a class="wiki" href="/wiki/Features/PreferenceSystem">Configuration and Preference System</a>
</li><li><a class="wiki" href="/wiki/Features/OpenSelection">Open Selection</a>
</li><li>Multi-document editing - as far as we understand this only means you can edit several files at the same time
</li><li>Code-assist editor - What does this cover?
</li><li>IME - <a class="wiki" href="/wiki/Features/MultilingualInputMethodEditor">Multilingual Input Method Editor</a>
</li><li><a class="wiki" href="/wiki/Features/IntegratedWebServer">Integrated Web Server</a> for web application development
</li><li>Call tips</li></ul>
<p>
* <a class="wiki" href="/wiki/Features/ToolsPrefernces">Tools Prefernces draft</a>
</p>
<h2 id="Perl-specificfeatures">Perl-specific features</h2>
<ul><li><a class="wiki" href="/wiki/Features/SyntaxChecking">Real Time Syntax Checking For Perl 5 and 6</a>
</li><li><a class="wiki" href="/wiki/Features/Perl5Debugger">Perl 5 Debugger</a> - perl debugging in Padre
</li><li><a class="wiki" href="/wiki/Features/RefactoringPerl5">Refactoring Perl 5 code</a>
</li><li><a class="wiki" href="/wiki/Features/AutomatedBuildAndTest">Automated Build and Test</a>
</li><li><a class="wiki" href="/wiki/Features/Perl6">Perl 6 features</a>
</li><li><a class="wiki" href="/wiki/Features/RegexEditor">Regex Editor</a>
</li><li><a class="wiki" href="/wiki/PluginAPI">Extendable in Perl 5</a> via a Powerful <a class="wiki" href="/wiki/PluginAPI">Plugin API</a>
</li><li><a class="wiki" href="/wiki/Features/PerlTidy">Features/PerlTidy</a> - automatic reformatting (tidying) of code
</li><li><a class="wiki" href="/wiki/Features/REPL">REPL</a>
</li><li><a class="wiki" href="/wiki/Features/ProjectBrowser">Project Browser</a> (Tree view of source files and resources)
</li><li><a class="wiki" href="/wiki/Features/IdentifyCommonErrors">Identify Common Errors</a>
</li><li>special features for beginners (we can create a page listing them)
</li><li>advanced features (we can create a page listing them)</li></ul>
<ul><li><a class="wiki" href="/wiki/Features/Extendable">Extendable in Perl 6 and other languages</a>
</li><li><a class="wiki" href="/wiki/Features/HandleMultipleLanguages">Features/HandleMultipleLanguages</a></li></ul>
<h2 id="Plugins">Plugins</h2>
<ul><li><a class="wiki" href="/wiki/Plugins">Plugins</a> generic plugins related tickets
</li><li><a class="wiki" href="/wiki/Features/IntegratedVersionControl">Integrated Version Control</a>
</li><li><a class="wiki" href="/wiki/Features/SuperGrep">Super Grep</a></li></ul>
<h1 id="Featurerequestsandideasfromothersources">Feature requests and ideas from other sources</h1>
<p>
In this section we have lists of features taken from verious articles
and web sites comparing tools and even from other IDEs.
</p>
<p>
For each feature we first link it to a related ticket but later
on we will point to the description of our implementation of
that feature.
</p>
<h2 id="FromPerlmonks">From Perlmonks</h2>
<p>
Perlmonks has a table comparing features of IDEs and editors to be used for Perl
development.
</p>
<p>
The table can be found at <a class="ext-link" href="http://www.perlmonks.org/?node=Perl%20Development%20Tools"><span class="icon"></span>Perl Development Tools</a>
</p>
<p>
TODO: add links to the relevant Feature Pages
</p>
<table>
<tr><td> <strong>Feature</strong>      </td><td> <strong>Status or link to ticket</strong> </td></tr>
<tr><td> Perl Exclusive     </td><td> Why is yes considered good here? </td></tr>
<tr><td> Free               </td><td> yes - Under the &quot;perl&quot; license. </td></tr>
<tr><td> Perl Engine        </td><td> Any - Padre itself runs on 5.8.7 or higher but your code can run on any version of perl </td></tr>
<tr><td> <a class="wiki" href="/wiki/Features/SyntaxChecking">Syntax Checking</a>  </td><td> yes </td></tr>
<tr><td> <a class="wiki" href="/wiki/Features/SyntaxHighlighting">Syntax colouring</a>  </td><td> yes </td></tr>
<tr><td> <a class="wiki" href="/wiki/Features/Autoindent">Autoindent</a>    </td><td> yes </td></tr>
<tr><td> <a class="wiki" href="/wiki/Features/IncrementalSearch">Incremental search</a> </td><td> yes </td></tr>
<tr><td> <a class="wiki" href="/wiki/Features/Brackets">Bracket Completition and Matching</a> </td><td> yes </td></tr>
<tr><td> Folding            </td><td> see under <a class="wiki" href="/wiki/Features/VisualEditorEffects">Visual Editor Effects</a> </td></tr>
<tr><td> Var tree           </td><td> see in <a class="wiki" href="/wiki/Features/Outline">Outline</a> </td></tr>
<tr><td> Code browser       </td><td> see in <a class="wiki" href="/wiki/Features/Outline">Outline</a> </td></tr>
<tr><td> Debugger           </td><td> <a class="wiki" href="/wiki/Features/Perl5Debugger">Perl 5 Debugger</a> </td></tr>
<tr><td> Stepping           </td><td> See Debugger </td></tr>
<tr><td> Breakpoints        </td><td> See Debugger </td></tr>
<tr><td> Watches            </td><td> See Debugger </td></tr>
<tr><td> Var evaluation     </td><td> See Debugger </td></tr>
<tr><td> Stack module       </td><td> See Debugger </td></tr>
<tr><td> <a class="wiki" href="/wiki/Features/Bookmarks">Bookmarks</a>    </td><td> yes </td></tr>
<tr><td> Suggest &amp; complete </td><td> see <a class="ticket" href="/ticket/67">#67</a> but add more </td></tr>
<tr><td> <a class="wiki" href="/wiki/Features/IdentifyCommonErrors">Identify errors</a> </td><td> also <a class="wiki" href="/wiki/Features/SyntaxChecking">Syntax Checking</a> </td></tr>
<tr><td> <a class="wiki" href="/wiki/Features/REPL">Interactive shell</a>  </td><td> </td></tr>
<tr><td> <a class="wiki" href="/wiki/Features/CodeSnippets">Code snippets</a>  </td><td> yes </td></tr>
<tr><td> <a class="wiki" href="/wiki/Features/ProjectManagement">Project</a> </td><td> </td></tr>
<tr><td> Control version    </td><td>  <a class="wiki" href="/wiki/Features/IntegratedVersionControl">Integrated Version Control</a> </td></tr>
<tr><td> Multi document     </td><td> yes </td></tr>
<tr><td> <a class="wiki" href="/wiki/Features/SplitView">Split View</a> </td><td> </td></tr>
<tr><td> IME                </td><td> Multilingual Input Method Editor </td></tr>
<tr><td> <a class="wiki" href="/wiki/Features/HelpViewer">Docs & Help viewer</a> </td><td> yes </td></tr>
<tr><td> <a class="wiki" href="/wiki/Features/CPANModuleInstaller">Module installer</a>   </td><td> </td></tr>
<tr><td> WEB                </td><td> <a class="wiki" href="/wiki/Features/IntegratedWebServer">Integrated Web Server</a> </td></tr>
<tr><td> FTP                </td><td> <a class="ticket" href="/ticket/12">#12</a> Read and write over ftp protocol (modify the program at remote server) </td></tr>
<tr><td> HTML export        </td><td> <a class="ticket" href="/ticket/112">#112</a> To export the source of program with the syntax colouring. </td></tr>
<tr><td> Linux              </td><td> <a class="wiki" href="/wiki/Download">Download</a> </td></tr>
<tr><td> BSD                </td><td> <a class="wiki" href="/wiki/DownloadFreeBSD">DownloadFreeBSD</a> and <a class="wiki" href="/wiki/DownloadNetBSD">DownloadNetBSD</a> </td></tr>
<tr><td> Mac OS X           </td><td> <a class="wiki" href="/wiki/DownloadMacOSX">DownloadMacOSX</a> </td></tr>
<tr><td> Solaris            </td><td> </td></tr>
<tr><td> Windows            </td><td> <a class="wiki" href="/wiki/[wiki:DownloadWindows]">[wiki:DownloadWindows]</a> </td></tr>
</table>
<h2 id="FromPerlNeedsBetterTools">From Perl Needs Better Tools</h2>
<p>
<a class="ext-link" href="http://www.eigenstate.net/perl_tools/"><span class="icon"></span>Perl Needs Better Tools</a> by Matisse Enzer 
</p>
<table>
<tr><td><strong>Feature</strong>                                                    </td><td> <strong>Status or link to ticket</strong> </td></tr>
<tr><td><a class="wiki" href="/wiki/Features/SyntaxHighlighting">Syntax coloring</a>               </td><td> </td></tr>
<tr><td><a class="wiki" href="/wiki/Features/SyntaxChecking">Real Time Syntax Checking</a> </td><td> </td></tr>
<tr><td><a class="wiki" href="/wiki/Features/IntegratedVersionControl">Version control integration</a> </td><td> </td></tr>
<tr><td>Code-assist editor                                               </td><td>  </td></tr>
<tr><td>Excellent refactoring support                                    </td><td> <a class="wiki" href="/wiki/Features/RefactoringPerl5">Refactoring Perl 5 code</a> </td></tr>
<tr><td>Tree view of source files and resources                          </td><td>  </td></tr>
<tr><td>Support for creating and running unit tests                      </td><td> <a class="wiki" href="/wiki/Features/AutomatedBuildAndTest">Automated Build and Test</a> </td></tr>
<tr><td>Language-specific help                                           </td><td> <a class="wiki" href="/wiki/Features/ContextSensitiveHelp">Context Sensitive Help</a> </td></tr>
<tr><td>Run your code under a debugger with real-time display of results </td><td>  </td></tr>
<tr><td>Automatic reformatting of code.                                  </td><td>  </td></tr>
<tr><td>handling of multiple languages                                   </td><td>  </td></tr>
<tr><td>Automated build and test                                         </td><td>  </td></tr>
</table>
<h2 id="FromEnlightenedPerl">From Enlightened Perl</h2>
<p>
Work together with the folks at Enlightened Perl <a class="ext-link" href="http://www.enlightenedperl.org/"><span class="icon"></span>http://www.enlightenedperl.org/</a>
to see how Padre can provide a GUI for their efforts.
Look at Kensho, <a class="ext-link" href="http://search.cpan.org/dist/Task-Kensho/"><span class="icon"></span>http://search.cpan.org/dist/Task-Kensho/</a> a list of recommended module
and see how Padre can provide a GUI (via a plugin or in core Padre) for them.
</p>
<h2 id="FromCompetitors">From Competitors</h2>
<p>
See the <a class="wiki" href="/wiki/'Competitor Analysis'">'Competitor Analysis'</a>
</p>
<h2 id="FromEclipseEPIC">From Eclipse EPIC</h2>
<p>
<a class="ext-link" href="http://www.eclipse.org/cdt/"><span class="icon"></span>Eclipse CDT - C/C++ Development Tooling</a>
<a class="ext-link" href="http://www.eclipse.org/jdt/"><span class="icon"></span>Eclipse Java development tools (JDT)</a>
</p>
<p>
Some of these features are only relevant (or possible) for C/C++ or Java
</p>
<table>
<tr><td><strong>Feature</strong>                                            </td><td> <strong>Status or link to ticket</strong> </td></tr>
<tr><td>Find all references                                      </td><td> </td></tr>
<tr><td>Find definition/declaration/usage                        </td><td> </td></tr>
<tr><td>Type Hierarchy                                           </td><td> </td></tr>
<tr><td>Call Hierarchy (or call graph)                           </td><td> </td></tr>
<tr><td><a class="wiki" href="/wiki/Features/RefactoringPerl5">Refactoring</a>             </td><td> </td></tr>
<tr><td><a class="wiki" href="/wiki/Features/SourceNavigation">Source navigation</a>       </td><td> </td></tr>
<tr><td><a class="wiki" href="/wiki/Features/SyntaxHighlighting">Syntax highlighting</a>   </td><td> </td></tr>
<tr><td>Code folding                                             </td><td> under <a class="wiki" href="/wiki/Features/VisualEditorEffects">Visual Editor Effects</a> </td></tr>
<tr><td>Hyperlink navigation                                     </td><td> </td></tr>
<tr><td>Include browser                                          </td><td> </td></tr>
<tr><td>Project creation                                         </td><td> <a class="wiki" href="/wiki/Features/ProjectSkeletonGeneration">Project Skeleton Generation</a> </td></tr>
<tr><td>Managed build for various toolchains                     </td><td> </td></tr>
<tr><td>Standard make build                                      </td><td> Via Run menu </td></tr>
<tr><td>Macro definition browser                                 </td><td> </td></tr>
<tr><td>Code generation                                          </td><td> partly (File -&gt; New...) </td></tr>
<tr><td>Visual debugging tools                                   </td><td> </td></tr>
<tr><td>Memory, registers, and disassembly viewers               </td><td> </td></tr>
</table>